external-crates/move/crates/move-analyzer/src/symbols.rs

```diff
@@ -1763,26 +1798,104 @@ impl Symbols {
 
 fn has_precompiled_deps(
     pkg_path: &Path,
-    pkg_dependencies: Arc<Mutex<BTreeMap<PathBuf, PrecomputedPkgDepsInfo>>>,
+    pkg_dependencies: Arc<Mutex<BTreeMap<PathBuf, PrecomputedPkgInfo>>>,
 ) -> bool {
     let pkg_deps = pkg_dependencies.lock().unwrap();
     pkg_deps.contains_key(pkg_path)
 }
 
+fn is_parsed_pkg_modified(
+    pkg_def: &P::PackageDefinition,
+    files_to_compile: &BTreeSet<PathBuf>,
+    file_hashes: Arc<BTreeMap<PathBuf, FileHash>>,
+) -> bool {
+    files_to_compile.iter().any(|fpath| {
+        if let Some(fhash) = file_hashes.get(fpath) {
+            if let P::Definition::Module(mdef) = &pkg_def.def {
+                return mdef.loc.file_hash() == *fhash;
+            }
+        }
+        false
+    })
+}
+
+fn is_typed_mod_modified(
+    mdef: &ModuleDefinition,
+    files_to_compile: &BTreeSet<PathBuf>,
+    file_hashes: Arc<BTreeMap<PathBuf, FileHash>>,
+) -> bool {
+    files_to_compile.iter().any(|fpath| {
+        if let Some(fhash) = file_hashes.get(fpath) {
+            return mdef.loc.file_hash() == *fhash;
+        }
+        false
```

Comment:
```suggestion
        file_hashes.get(fpath)
    .map(|fhash| mdef.loc.file_hash() == *fhash)
    .unwrap_or(false)
```

Reply:
Thanks! An arguably even more Rust-y one would be:
```rust
file_hashes
  .get(fpath)
  .map_or(false, |fhash| mdef.loc.file_hash() == *fhash)
```